metacat-main/src/main/java/com/netflix/metacat/main/services/impl/ConnectorTableServiceProxy.java

```diff
@@ -0,0 +1,229 @@
+     * @return list of table names
+     */
+    public List<QualifiedName> getQualifiedNames(final String uri, final boolean prefixSearch) {
+        final List<QualifiedName> result = Lists.newArrayList();
+
+        connectorManager.getTableServices().forEach(service -> {
+            final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+            final ConnectorRequestContext connectorRequestContext
+                = converterUtil.toConnectorContext(metacatRequestContext);
+            try {
+                final Map<String, List<QualifiedName>> names =
+                    service.getTableNames(connectorRequestContext, Lists.newArrayList(uri), prefixSearch);
+                final List<QualifiedName> qualifiedNames = names.values().stream().flatMap(Collection::stream)
+                    .collect(Collectors.toList());
+                result.addAll(qualifiedNames);
+            } catch (final UnsupportedOperationException uoe) {
+                log.debug("Table service doesn't support getting table names by URI. Skipping");
+            }
+        });
+        return result;
+    }
+
+    /**
+     * Calls the connector table service getTableNames method.
+     * @param uris list of locations
+     * @param prefixSearch if false, the method looks for exact match for the uri
+     * @return list of table names
+     */
+    public Map<String, List<QualifiedName>> getQualifiedNames(final List<String> uris, final boolean prefixSearch) {
+        final Map<String, List<QualifiedName>> result = Maps.newHashMap();
+
+        connectorManager.getTableServices().forEach(service -> {
+            final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
+            final ConnectorRequestContext connectorRequestContext
+                = converterUtil.toConnectorContext(metacatRequestContext);
+            try {
+                final Map<String, List<QualifiedName>> names =
+                    service.getTableNames(connectorRequestContext, uris, prefixSearch);
+                names.forEach((uri, qNames) -> {
+                    final List<QualifiedName> existingNames = result.get(uri);
+                    if (existingNames == null) {
+                        result.put(uri, qNames);
+                    } else {
+                        existingNames.addAll(qNames);
+                    }
+                });
+            } catch (final UnsupportedOperationException uoe) {
+                log.debug("Table service doesn't support getting table names by URI. Skipping");
+            }
+        });
+        return result;
+    }
+
+    /**
+     * Calls the connector table service exists method.
+     * @param name table name
+     * @return true, if the object exists.
+     */
+    public boolean exists(final QualifiedName name) {
+        final MetacatRequestContext metacatRequestContext = MetacatContextManager.getContext();
```

Comment:
should we bring the table to the cache for the exist() call?

Reply:
In our initial effort, we use cache only during the getTable API. Later on we can expand it to other internal apis. We can add the support to exists call as well.